File: redbook-fronted/src/api/product.ts
import request from './request';

export interface Product {
    id?: number;
    name: string;
    price: number;
    image: string;
    time?: string;
    sales?: number;
    userId?: number;
    shopId?: number;
    stock: number;
}

export interface ProductVo extends Product {
    shop?: any;
    customAttributes?: any[];
}

export interface ProductSearchDto {
    pageNum: number;
    pageSize: number;
    keyword?: string;
    minPrice?: number;
    maxPrice?: number;
    shopId?: number;
}

export const getProductList = (pageNum: number = 1, pageSize: number = 10) => {
    return request({
        url: '/product/getProductList',
        method: 'get',
        params: { pageNum, pageSize }
    });
};

export const getProduct = (id: number) => {
    return request({
        url: `/product/getProduct/${id}`,
        method: 'get'
    });
};

export const postProduct = (data: Product) => {
    return request({
        url: '/product/postProduct',
        method: 'post',
        data
    });
};

export const updateProduct = (data: Product) => {
    return request({
        url: '/product/updateProduct',
        method: 'put',
        data
    });
};

export const deleteProduct = (productId: number) => {
    return request({
        url: `/product/delete/${productId}`,
        method: 'delete'
    });
};

export const searchProduct = (data: ProductSearchDto) => {
    return request({
        url: '/product/search',
        method: 'post',
        data
    });
};

File: redbook-service/redbook-service-order/src/main/java/com/itcast/model/pojo/Product.java
package com.itcast.model.pojo;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

import java.io.Serializable;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class Product implements Serializable {
    private static final long serialVersionUID = 1L;

    /**
     * 主键
     */
    private Integer id;

    /**
     * 分类id（商品服务返回的字段）
     */
    private Integer categoryId;

    /**
     * 商品名称
     */
    private String name;

    /**
     * 商品价格
     */
    private Double price;

    /**
     * 商品图片
     */
    private String image;

    private String mainImage;

    /**
     * 发布时间
     */
    private String time;

    /**
     * 销量
     */
    private Integer sales;

    /**
     * 用户id
     */
    private Integer userId;

    /**
     * 店铺id
     */
    private Integer shopId;

    /**
     * 库存
     */
    private Integer stock;
}
